kafka-ui-react-app/src/lib/permissions.ts

```diff
@@ -83,32 +91,27 @@ export function isPermitted({
   if (!clusterMap) return false;
 
   // short circuit
-  const resourceData = clusterMap.get(resource);
-  if (!resourceData) return false;
-
-  return (
-    resourceData.findIndex((item) => {
-      let valueCheck = true;
-      if (item.value) {
-        valueCheck = false;
-
-        if (value) valueCheck = new RegExp(item.value).test(value);
-      }
-
-      // short circuit
-      if (!valueCheck) return false;
-
-      if (!Array.isArray(action)) {
-        return item.actions.includes(action);
-      }
-
-      // every given action should be found in that resource
-      return action.every(
-        (currentAction) =>
-          item.actions.findIndex((element) => element === currentAction) !== -1
-      );
-    }) !== -1
-  );
+  const resourcePermissions = clusterMap.get(resource);
+  if (!resourcePermissions) return false;
+
+  const valueMatches = (
```

Comment:
we can put this function outside of this function , so we will not create a new reference everytime.

Reply:
Done, thanks